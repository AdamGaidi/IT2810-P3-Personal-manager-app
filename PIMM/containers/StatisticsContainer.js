import React, { Component } from "react";
import { ScrollView, StyleSheet, View, Text, Animated } from "react-native";
import Colors from "../constants/Colors";
import Strings from "../constants/Strings";
import StatisticsItem from "../components/StatisticsItem";

export default class StatisticsContainer extends Component {
  render() {
    const { stepsToday, stepsThisWeek } = this.props;
    return (
      <View style={styles.statisticsContainer}>
        <Text style={styles.headerText}>Statistics</Text>
        <View />
        <StatisticsItem text={Strings.statistics.today} steps={stepsToday} />
        <StatisticsItem
          text={Strings.statistics.average}
          steps={Math.round(stepsThisWeek / 7)}
        />
        <StatisticsItem
          text={Strings.statistics.lastWeek}
          steps={stepsThisWeek}
        />
        <View />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  headerText: {
    marginBottom: 20,
    color: Colors.primaryBlue,
    fontSize: 18,
    lineHeight: 19
  },
  statisticsContainer: {
    marginTop: 30
  }
});
