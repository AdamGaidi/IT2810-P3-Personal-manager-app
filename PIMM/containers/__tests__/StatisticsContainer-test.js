import React from "react";
import StatisticsContainer from "../StatisticsContainer";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders shallow correctly with example props", () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <StatisticsContainer stepsToday={5000} stepsThisWeek={10000} />
  );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
