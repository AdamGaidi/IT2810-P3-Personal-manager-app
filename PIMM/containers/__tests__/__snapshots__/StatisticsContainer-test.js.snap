// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shallow correctly with example props 1`] = `
<Component
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5a95ff",
        "fontSize": 18,
        "lineHeight": 19,
        "marginBottom": 20,
      }
    }
  >
    Statistics
  </Text>
  <Component />
  <StatisticsItem
    steps={5000}
    text="Today"
  />
  <StatisticsItem
    steps={1429}
    text="Daily average last 7 days"
  />
  <StatisticsItem
    steps={10000}
    text="Total last 7 days"
  />
  <Component />
</Component>
`;
